File: src/CorporateMessengerWithAgile/Application/Dto/KanbanBoardColumnDto.cs
using Domain.Entity;

namespace Application.Dto
{
    public class KanbanBoardColumnDto : BaseDto
    {
        public Guid TeamId { get; set; }
        public TaskItemStatus TaskStatus { get; set; }
        public int PositionOnBoard { get; set; }
        public string Title { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/KanbanBoardColumnDto.ts
// src/models/entity/KanbanBoardColumnDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const KanbanBoardColumnDtoSchema = BaseDtoSchema.extend({
    teamId: GuidSchema,
    taskStatus: z.string().min(1),
    positionOnBoard: z.number(),
    title: z.string().min(1),
});

export type KanbanBoardColumnDto = z.infer<typeof KanbanBoardColumnDtoSchema>;
